### ISSUE TITLE
Add AuthN Policy assignment to Bookmark and SWA app resources

### ISSUE THREAD

**Issue by sgal-dm** [okta/terraform-provider-okta#1203]
### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

All application types can be associated with an Authentication Policy, but only the _okta_app_saml_ and _okta_app_oauth_ resources have the necessary _authentication_policy_ attribute to build that association with Terraform.

Looking at [API documentation](https://developer.okta.com/docs/reference/api/apps/#add-application), it seems like this could be implemented across the board. Can the authentication_policy attribute be added to Bookmark and SWA application resources?

### New or Affected Resource(s)

- okta_app_bookmark
- okta_app_swa

### Potential Terraform Configuration

```
# Retrieve the authentication policy
data "okta_policy" "auth" {
  type  = "ACCESS_POLICY"
  name  = "non-default"
}

resource "okta_app_bookmark" "bookmark" {
  ...
  authentication_policy = data.okta_policy.auth.id
  ...
}

resource "okta_app_swa" "swa" {
  ...
  authentication_policy = data.okta_policy.auth.id
  ...
}
```

### References

- [okta_app_saml](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_saml) resource documentation. This is already implemented for this resource type.
- [okta_app_bookmark](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_bookmark) and [okta_app_swa](https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_swa) resource documentation, not implemented for these resources.
- [Apps API documentation](https://developer.okta.com/docs/reference/api/apps/#add-application).

**Comment by tgoodsell-tempus** [comment]
I'm against adding these directly to the resource due to https://github.com/okta/terraform-provider-okta/issues/1202 and the terraform best practices (https://www.terraform.io/plugin/hashicorp-provider-design-principles#resources-should-represent-a-single-api-object).

However, I think creating a standalone "link" resource that works generically for all apps is do able and the better route. I can plan on spending some time making this within the next couple weeks, if I can get my hands on a OIE dev environment.

**Comment by monde** [comment]
We'll have to prioritize working on this.

**Comment by monde** [comment]
Okta internal reference: https://oktainc.atlassian.net/browse/OKTA-524634

**Comment by sgal-dm** [comment]
Not Stale

**Comment by gleeblezoid** [comment]
Agreed, not stale, and it'd actually be great to include `okta_app_auto_login` apps in this too (or I'm happy to raise a separate issue for that - whichever folks prefer).

**Comment by danielpops** [comment]
Any chance this would get worked on? It's not great that we can set the authentication_policy for oauth, saml, and bookmark apps but not swa or auto_login apps.

**Comment by danielpops** [comment]
oh! I just learned about https://registry.terraform.io/providers/okta/okta/latest/docs/resources/app_access_policy_assignment which I guess will work for what I want to do.
